### PR TITLE
Add model checkers based on `astroid` nodes

### PR DESCRIPTION
**Pull request by yalef**
Resolves: #85
Resolves: #94
Resolves: #95

Default python [ast](https://docs.python.org/3/library/ast.html) module doesn't provide methods to dive into class inheritance tree, it can only check names of base classes. 
There're extended ast variant for python and it's called [astroid](https://github.com/pylint-dev/astroid). There're interesting feature called [inference](https://pylint.readthedocs.io/projects/astroid/en/latest/inference.html#). And by using it we can dive into class inheritance tree and say is it Django model or not (find `django.db.models.Model` there). 

* all model checkers was rewrited to work with astroid nodes to be able to check their inheritance tree.
* `django` package added to dev dependencies because it needed for proper work of tests
* minimal version of `python` increased to `3.8` due `django` and `astroid` dependencies
* add new test for case with abstract base models
* add new test for `ModelForm` case from #95 issue

### CODE REVIEW

**Comment by rocioar** [comment]
Thank you, this is great! Looks like one of the test jobs is failing, specifically the one for 3.7, could you take a looK?

File: pyproject.toml [hunk 1]
```diff
@@ -23,10 +23,12 @@ requires = ["poetry_core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+python = "^3.8"
```
Comment (rocioar) [inline pyproject.toml:26]: Is astroid not supported in version 3.7?

Comment (yalef) [reply pyproject.toml]: python 3.8 was required by django 4.2 which was added to dev dependencies(for testing). I reduced the version of django to 3.2 and the python version to 3.7.2(minimal version required for astroid)

File: pyproject.toml [hunk 2]
```diff
@@ -23,10 +23,12 @@ requires = ["poetry_core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.poetry.dependencies]
-python = "^3.7"
+astroid = "^2.15.2"
+python = "^3.7.2"
```
Comment (rocioar) [inline pyproject.toml:27]: Could the `.2` part be removed?

Comment (yalef) [reply pyproject.toml:27]: Unfortunately no. Minimal python version for `astroid` is `3.7.2`

**Comment by ron8mcr** [comment]
Hi @rocioar, is there any chance for this to be merged soon?

**Pull request merged**